consuming-webservices/ViewModel.swift

```diff
@@ -44,14 +44,19 @@ class ViewModel {
         NetworkRequest(path: colorUrl, method: .get).execute { result in
             switch result {
             case .success(let data):
-                guard
-                    let data = data as? [String: Any],
-                    let color = data["color"] as? String
-                    else { return }
-                let currentColor = UIColor(color)
-                self.color = currentColor
+                guard let jsonData = data as? Data else {
+                    return
+                }
+                do {
+                    let colorData = try JSONDecoder().decode(ColorData.self, from: jsonData)
+                    let currentColor = UIColor(colorData.color)
+                    self.color = currentColor
```

Comment:
For consistency either directly assign `UIColor(colorData.color)` to self.color like in `setColor` or add `currentColor` to `setColor`